mapic/mistapiconnector_app.go

```diff
@@ -633,36 +686,46 @@ func (mc *mac) refreshStreamInfo(playbackID string) (*streamInfo, error) {
 		newPushes[pushURL] = &pushStatus{
 			target:  target,
 			profile: ref.Profile,
-			// Assume setup was all successful
-			pushStartEmitted: true,
-			metrics:          &data.MultistreamMetrics{},
-		}
-		if infoExists {
-			info.mu.Lock()
-			push, pushExists := info.pushStatus[pushURL]
-			info.mu.Unlock()
-			if pushExists {
-				newPushes[pushURL].metrics = push.metrics
-				newPushes[pushURL].pushStopped = push.pushStopped
-				newPushes[pushURL].pushStartEmitted = push.pushStartEmitted
-			}
+			metrics: &data.MultistreamMetrics{},
 		}
 	}
 
-	newInfo := &streamInfo{
-		id:         stream.ID,
-		stream:     stream,
-		isLazy:     true, // flag it as a lazy stream info to avoid sending metrics
-		done:       make(chan struct{}),
-		pushStatus: newPushes,
-		// Assume setup was all successful
-		multistreamStarted: true,
+	mc.mu.Lock()
+	defer mc.mu.Unlock()
+	info, exists := mc.streamInfo[playbackID]
+	if !exists {
+		info = &streamInfo{
+			id:         stream.ID,
+			stream:     stream,
+			isLazy:     true, // flag it as a lazy stream info to avoid sending metrics
+			done:       make(chan struct{}),
+			pushStatus: newPushes,
+		}
+		mc.streamInfo[playbackID] = info
+	} else {
+		info.id = stream.ID
+		info.stream = stream
+	}
+	info.mu.Lock()
+	defer info.mu.Unlock()
+
+	for newPushURL, newPush := range newPushes {
+		if push, exists := info.pushStatus[newPushURL]; exists {
+			push.mu.Lock()
+			push.target = newPush.target
+			push.profile = newPush.profile
```

Comment:
Interesting, does the reconcile logic also take care of changing profiles in case the config changed? Also, are the pushes reconciled automatically after this function runs?

Reply:
No, it does not. But since we're refreshing the whole stream here, I think it's worth also refreshing the target profile.